lib/logstash/inputs/http.rb

```diff
@@ -42,6 +42,10 @@ class LogStash::Inputs::Http < LogStash::Inputs::Base
   # Set the truststore password
   config :keystore_password, :validate => :password
 
+  # here you can describe how to decode specific content-types
+  # by default the default codec will be used
+  config :additional_codecs, :validate => :hash, :default => { "application/json" => "json" }
```

Comment:
I think this can evolve in a bwc way into 

simple case (as-is): `{ "application/json" => "json" }`

or more complex:  `{ "application/json" => { "json" => { codec_option => "meh" } }`